code/game/objects/items/religion.dm

```diff
@@ -1,27 +1,126 @@
 /obj/item/banner
 	name = "banner"
+	desc = "A banner with Nanotrasen's logo on it."
 	icon = 'icons/obj/items_and_weapons.dmi'
 	icon_state = "banner"
 	item_state = "banner"
+	force = 8
+	attack_verb = list("forcefully inspired", "violently encouraged", "relentlessly galvanized")
 	lefthand_file = 'icons/mob/inhands/equipment/banners_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/banners_righthand.dmi'
-	desc = "A banner with Nanotrasen's logo on it."
-	var/moralecooldown = 0
-	var/moralewait = 600
+	var/morale_time = 0
+	var/morale_cooldown = 600 //How many deciseconds between uses
+	var/list/job_loyalties //Mobs with any of these assigned roles will be inspired
+	var/list/role_loyalties //Mobs with any of these special roles will be inspired
+	var/warcry
 
 /obj/item/banner/attack_self(mob/living/carbon/human/user)
-	if(moralecooldown + moralewait > world.time)
+	if(morale_time > world.time)
+		to_chat(user, "<span class='warning'>You aren't feeling inspired enough to flourish [src] again yet.")
 		return
-	to_chat(user, "<span class='notice'>You increase the morale of your fellows!</span>")
-	moralecooldown = world.time
-
-	for(var/mob/living/carbon/human/H in range(4,get_turf(src)))
-		to_chat(H, "<span class='notice'>Your morale is increased by [user]'s banner!</span>")
-		H.adjustBruteLoss(-15)
-		H.adjustFireLoss(-15)
-		H.AdjustStun(-40)
-		H.AdjustKnockdown(-40)
-		H.AdjustUnconscious(-40)
+	user.visible_message("<span class='big notice'>[user] flourishes [src]!</span>", \
+	"<span class='notice'>You raise [src] skywards, inspiring your allies!</span>")
+	playsound(user, "rustle", 100, FALSE)
+	if(warcry)
+		user.say("[warcry]!!")
+	user.transform *= 1.2
+	animate(user, transform = initial(user.transform), time = 10)
```

Comment:
Make a copy of the transform before you change it and animate to that instead of initial()